File: problem9/python/problem9.py
import math
def numberSqrt(x,y):
    result = math.hypot(x,y)
    return  result
def pythagoreanTriplet():
    for i in range(1,1000):
        for j in range(1,1000):
            if numberSqrt(i,j) + i + j == 1000:
                return i,j,int(numberSqrt(i,j))
def productABC():
    return reduce(lambda x,y: x*y, pythagoreanTriplet())
print productABC()
            

File: problem9/java/problem9.java
/**
 * Created by Administrator on 2014/12/15.
 */
public class Problem9 {
    public static int pythagoreanTriplet() {
        for (int i = 1; i < 2000; i++) {
            for (int j = 1; j < 2000; j++) {
                if ((i + j + Math.hypot(i, j)) == 1000) {      /*此处只能让他自动往精度更细小的方向转换，否则可能不正确*/
                    return i * j * (int)Math.hypot(i, j);
                }
            }
        }
        return 0;
    }

    public static void main(String[] args) {
        int result = pythagoreanTriplet();
        System.out.println(result);
    }
}
